### PR TITLE
Drop version tags for vault and vault-k8s

### PR DESCRIPTION
**Pull request by Dentrax**
Drop version tags for `vault` and `vault-k8s` images.

### CODE REVIEW

**Review by imjasonh — APPROVED** [review]

File: images/vault-k8s/config/template.apko.yaml [hunk 1]
```diff
@@ -1,14 +1,12 @@
 contents:
   packages:
-    - vault-k8s
-    - libcap
 
 accounts:
   groups:
-    - groupname: vault
+    - groupname: vault-k8s
```
[the remark this answers is not in the document]
Comment (Dentrax) [reply images/vault-k8s/config/template.apko.yaml:6]: Actually we don't. But it'd be great to make those consistent with the package name, as we did follow similar way for other packages.

Maybe we should refactor it for all the images and just set: `nonroot`. 🤷

[the remark this answers is not in the document]
Comment (imjasonh) [reply images/vault-k8s/config/template.apko.yaml:6]: I'd like to make as few changes as possible in here, and consider renaming the user as a separate change. Maybe it should just be named `nonroot` and we should have better processes/linters in place to encourage that standard.

[the remark this answers is not in the document]
Comment (Dentrax) [reply images/vault-k8s/config/template.apko.yaml:6]: > Maybe it should just be named nonroot and we should have better processes/linters in place to encourage that standard.

+1, can i file a proposal for this? to enforce `nonroot` everywhere. if accepted, then i can create a linter rule for this.

[the remark this answers is not in the document]
Comment (Dentrax) [reply images/vault-k8s/config/template.apko.yaml]: Reverted back to `vault`

File: images/vault/tests/main.tf [hunk 2]
```diff
@@ -4,18 +4,11 @@ terraform {
...
 }
 
-variable "digests" {
-  description = "The image digests to run tests over."
-  type = object({
-    vault     = string
-    vault-k8s = string
-  })
+variable "digest" {
+  description = "The image digest to run tests over."
```
Comment (mattmoor) [inline images/vault/tests/main.tf:8]: This is a breaking change for us downstream.  We discussed on slack keeping this as a single module, let's do that unless we have a really good reason not to.

**Review by mattmoor — CHANGES_REQUESTED** [review]

File: images/vault/tests/main.tf (hunk 2 above)
Comment (Dentrax) [reply images/vault/tests/main.tf]: Done!

File: images/vault/vault.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module "vault-config" { source = "./configs/vault" }
+
+module "vault" {
+  source            = "../../tflib/publisher"
+  name              = basename(path.module)
+  target_repository = var.target_repository
```
Comment (Dentrax) [inline images/vault/vault.tf:6]: is it correct? or should we have `-vault` suffix here?

File: images/vault/main.tf [hunk 4]
```diff
@@ -1,53 +1,38 @@
...
-    vault     = "vault-1.14",
-    vault-k8s = "vault-k8s",
-  }
-
-  repositories = {
-    vault     = var.target_repository,
-    vault-k8s = "${var.target_repository}-k8s",
+terraform {
+  required_providers {
+    apko = { source = "chainguard-dev/apko" }
```
Comment (mattmoor) [inline images/vault/main.tf:3]: you don't need this provider

File: images/vault/vault.tf (hunk 3 above)
Comment (mattmoor) [reply images/vault/vault.tf:6]: This is perfect otherwise it is `vault-vault`

File: images/vault/main.tf [hunk 5]
```diff
@@ -1,53 +1,38 @@
...
+    oci  = { source = "chainguard-dev/oci" }
```
Comment (mattmoor) [inline images/vault/main.tf:4]:
```suggestion
    oci = { source = "chainguard-dev/oci" }
```

**Review by mattmoor — APPROVED** [review]

**Pull request merged**